src/binder/bind/bind_projection_clause.cpp

```diff
@@ -251,33 +250,14 @@ bool Binder::isOrderByKeyTypeSupported(const LogicalType& dataType) {
     return true;
 }
 
-uint64_t Binder::bindSkipLimitExpression(const ParsedExpression& expression) {
+std::shared_ptr<Expression> Binder::bindSkipLimitExpression(const ParsedExpression& expression) {
     auto boundExpression = expressionBinder.bindExpression(expression);
-    auto errorMsg = "The number of rows to skip/limit must be a non-negative integer.";
-    if (boundExpression->expressionType != ExpressionType::LITERAL) {
-        throw BinderException(errorMsg);
+    if (boundExpression->expressionType != ExpressionType::LITERAL &&
```

Comment:
I have a wild idea.

Can we keep limit/skip number as `int64_t`. 

When expr is of type LITERAL, we set skip/limit properly.
When expr is of type PARAMETER, if PARAMETER is set, we set skip/limit property. If not set, we set them to 0 or 1 or some constant. Since we will always rebind with prepare this seems fine?

Also we should probably also support when expr is a foldable constant, something like 1 + 2